pandas/core/construction.py

```diff
@@ -541,6 +528,37 @@ def sanitize_array(
     return subarr
 
 
+def _sanitize_str_dtypes(
```

Comment:
intent is to try to deduplicate with other things we have in core/dtypes/cast ?

Reply:
partially.  more immediately i'm focused on trying to isolate scalar cases so we can rule them out elsewhere